lib/_stream_writable.js

```diff
@@ -424,12 +424,22 @@ function onwriteError(stream, state, sync, er, cb) {
     // this can emit finish, and it will always happen
     // after error
     process.nextTick(finishMaybe, stream, state);
+
+    // needed for duplex, fixes https://github.com/nodejs/node/issues/6083
+    if (stream._readableState) {
```

Comment:
I haven't looked in enough detail but I guess there's no way to do this in `_stream_duplex.js` instead?

Reply:
My goal would be to change this into `stream.destroy(err)`. But that's a more massive change, see https://github.com/nodejs/node/issues/20096.

Reply:
Nit: can't we only 1 one of these and put it outside the wrapper `if` instead of duplicating it?

Reply:
we can't as the two branches are significantly different. However we can move it in its own function. Should I do that?

Reply:
In the second block, you are setting `errorEmitted` before calling the callback. I remember spending a significant amount of time on this block of code, and these two paths should remain separated. It's something I'd prefer not to refactor in this PR.

Reply:
Yes `stream._writableState.errorEmitted = true` can stay as is if the user provided callback needs to find it set to `false` (why?) but I see no reason to not move the other part. There can't be existing code that relies on new code.

Anyway it's a just a nit and it can be ignored.